lib/std/fs.zig

```diff
@@ -2232,6 +2240,50 @@ pub fn selfExePath(out_buffer: []u8) SelfExePathError![]u8 {
             // TODO could this slice from 0 to out_len instead?
             return mem.spanZ(@ptrCast([*:0]u8, out_buffer));
         },
+        .openbsd => {
+            // OpenBSD doesn't support getting the path of a running process, so try to guess it
+            if (os.argv.len >= 1) {
+                if (str_containsZ(os.argv[0], '/')) {
+                    // argv[0] is a path (relative or absolute): use realpath(3) directly
+                    var real_path_buf: [MAX_PATH_BYTES]u8 = undefined;
+                    const real_path = try os.realpathZ(os.argv[0], &real_path_buf);
+                    if (real_path.len > out_buffer.len)
+                        return error.NameTooLong;
+                    mem.copy(u8, out_buffer, real_path);
+                    return out_buffer[0..real_path.len];
+
+                } else if (os.argv[0][0] != '\x00') {
+                    // argv[0] is not empty (and not a path): search it inside PATH
+                    const paths = std.os.getenv("PATH") orelse "";
+                    var path_it = mem.split(paths, ":");
+                    while (path_it.next()) |a_path| {
+                        var resolved_path_buf: [MAX_PATH_BYTES:0]u8 = undefined;
+                        const resolved_path = std.fmt.bufPrint(&resolved_path_buf, "{}/{}\x00", .{
+                            a_path,
+                            os.argv[0],
+                        }) catch "";
+
+                        var real_path_buf: [MAX_PATH_BYTES:0]u8 = undefined;
+                        if (os.realpathZ(&resolved_path_buf, &real_path_buf) catch null) |real_path| {
+                            // found a file, and hope it is the right file
+                            if (real_path.len > out_buffer.len)
+                                return error.NameTooLong;
+                            mem.copy(u8, out_buffer, real_path);
+                            return out_buffer[0..real_path.len];
+                        }
+                    }
+                }
+            }
+
+            if (os.getenv("_")) |sh_exefile| {
+                // sh (or bash) sets "_" environment variable
```

Comment:
sets it to what? (TIL: the exe it invoked)

However this would be inherited by child processes; so seems like a bad idea?